internal/pkg/siderolink/provision.go

```diff
@@ -398,15 +487,63 @@ func (h *ProvisionHandler) buildProvisionContext(ctx context.Context, req *pb.Pr
 		return nil, err
 	}
 
+	if uniqueToken := pointer.SafeDeref(req.NodeUniqueToken); uniqueToken != "" {
+		requestNodeUniqueToken, err = jointoken.ParseNodeUniqueToken(uniqueToken)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if link != nil {
+		_, forceValidUniqueToken = link.Metadata().Annotations().Get(siderolink.ForceValidNodeUniqueToken)
+
+		linkNodeUniqueToken, err = jointoken.ParseNodeUniqueToken(link.TypedSpec().Value.NodeUniqueToken)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	machineStatus, err := safe.ReaderGetByID[*system.ResourceLabels[*omni.MachineStatus]](ctx, h.state, req.NodeUuid)
```

Comment:
We can use InfraMachineStatus last wipe time?

Change the infra machine status resource to have the last wiped token value.
Populate it this way Link -> InfraMachine -> InfraMachineStatus.